#Moving Obstacle #1

#Number of orders:
#Number of shipments per order: 1
#Challenges:
#  -Interruption: No
#  -Drop parts: No
#  -Sensor blackout: No
#  -Part flipping: No
#  -Faulty parts: Yes
#  -Moving obstacle: No
#Containers
# -Belt: No
# -Bins: Optional
# -Shelf: no
#AGV
# -order_0:
#   -shipment_0: agv1
# -order_1:
#   -shipment_0: agv2
#Time limit: No

time_limit: 500

options:
  gazebo_state_logging: true
  insert_models_over_bins: false
  insert_models_over_shelves: true
  belt_population_cycles: 0
  spawn_extra_models: false
  model_type_aliases:
    order_part1: piston_rod_part_blue
    order_part2: gasket_part_red
    order_part3: gasket_part_green

shelf_layout:
  row_1: ['base',0,'base','base'] #shelves: 3, 4, 5
  row_2: ['pipe','base',0,'base']   #shelves: 6, 7, 8
  row_3: ['base','pipe',0,'collar'] #shelves: 9, 10, 11

aisle_layout:
  #Aisle 2
  person_1:
    location: 1.57
    start_time: 16.
    move_time: 9.
    wait_time: 7.

#--Config 2
orders:
  order_0:
    announcement_condition: time
    announcement_condition_value: 0.0
    shipment_count: 1
    destinations: [agv1]
    products:
#      part_0:
#        type: piston_rod_part_blue
#        pose:
#          xyz: [0.15, 0.2, 0]
#          rpy: [0, 0, 'pi/4']
#      part_1:
#        type: gasket_part_red
#        pose:
#          xyz: [0.15, 0.2, 0]
#          rpy: [0, 0, '-pi/4']
      part_2:
        type: gasket_part_green
        pose:
          xyz: [-0.15, -0.2, 0]
          rpy: [0, 0, '-pi/4']

#2 arms advantageous
models_over_shelves:
  shelf11:
    models:
      gasket_part_green:
        xyz_start: [1, 0.45, 0.0]
        xyz_end: [0., 0.45, 0]
        rpy: [0, 0, 'pi/4']
        num_models_x: 8
        num_models_y: 1
  shelf5:
    models:
      piston_rod_part_blue:
        xyz_start: [1, 0.45, 0.0]
        xyz_end: [0., 0.45, 0.0]
        rpy: [0, 0, 'pi/4']
        num_models_x: 8
        num_models_y: 1

  shelf8:
    models:
      gasket_part_red:
        xyz_start: [1, -0.45, 0.0]
        xyz_end: [0., -0.45, 0.0]
        rpy: [0, 0, 'pi/4']
        num_models_x: 8
        num_models_y: 1

#Faulty products
faulty_products:
  - piston_rod_part_blue_1
  - gasket_part_red_1
  - gasket_part_green_1
  - piston_rod_part_blue_3
  - gasket_part_red_2
  - gasket_part_green_3
